### ISSUE TITLE
Looping Command Error: threw the following error: substring not found

### ISSUE THREAD

**Issue by rnjv** [Torantulino/Auto-GPT#761]
### Duplicates

- [X] I have searched the existing issues

### Steps to reproduce 🕹

I asked to parse github url as a code assistant.

### Current behavior 😯

```
line 51, in fix_and_parse_json brace_index = json_str.index("{") ^^^^^^^^^^^^^^^^^^^ ValueError: substring not found
NEXT ACTION:  COMMAND = Error: ARGUMENTS = substring not found
SYSTEM:  Command Error: threw the following error: substring not found
```

### Expected behavior 🤔

scrape the modules properly

### Your prompt 📝

Name:  foo-GPT
Role:  to autonomously develop a codebase for X
Goals: ['Use Y library here to improve the outdated code here: X, 'Use foo,bar or other optimizations for making the code faster to run.', 'Create classes and modules for specific functions X, Y and Z. ', 'Save the module in a folder and let me know.']

**Comment by FlyGuyGo** [comment]
I have this error as well.

**Comment by wantedrbk** [comment]
the same problem

**Comment by wanjianwei** [comment]
+1

**Comment by willmedorneles** [comment]
Same error here

**Comment by jfgosselin** [comment]
+1

**Comment by Gbuz** [comment]
I was getting a lot of Error: Invalid JSON So i just made a prompt to get it to just forget about it and move on if it got stuck :D

"The program should be designed to handle errors and exceptions that may occur during its operation. If the program encounters a JSON error, it should be able to move on and continue with its tasks without causing a complete system failure. However, if the error is something that the program can solve or recover from, it should do so to ensure that it completes the task accurately and efficiently. This can be achieved by implementing error-handling mechanisms and integrating them into the program’s decision-making process. By doing so, the program can ensure that it operates reliably and effectively, even when faced with unexpected errors or exceptions."

it's been working for over and hour so far no errors.

**Comment by sswam** [comment]
It doesn't seem to like apostrophes in the user input, that's one thing that can trigger this error, I think.

**Comment by sswam** [comment]
Using JSON is a bad design decision, it would be better to use a more flexible format which the AI can't get wrong, such as plain text with headings, and each section optional; or some sort of non-strict YAML.

**Comment by Torantulino** [comment]
Fixed in #759

**Comment by Torantulino** [comment]
@sswam Agreed, it would be great to get something like this set up to compare
